xcode/Subconscious/Shared/Components/AppView.swift

```diff
@@ -2896,9 +2896,19 @@ struct AppModel: ModelProtocol {
     }
 }
 
+protocol TestableEnvironment {
+    var noosphere: NoosphereService { get }
+    var database: DatabaseService { get }
+    var data: DataService { get }
+    var transclude: TranscludeService { get }
+    
+    var addressBook: AddressBookService { get }
+    var userProfile: UserProfileService { get }
+}
```

Comment:
Introduced this so that we can have static functions that work in both testing and the app itself. 

We could alternatively move `DetailStackModel.findBestAddressForLink` into one a service to avoid this change.